source/extensions/filters/http/ext_proc/http_client/http_client_impl.cc

```diff
@@ -110,18 +109,24 @@ void ExtProcHttpClient::onFailure(const Http::AsyncClient::Request&,
   ASSERT(reason == Http::AsyncClient::FailureReason::Reset ||
          reason == Http::AsyncClient::FailureReason::ExceedResponseBufferLimit);
   ENVOY_LOG(error, "Request failed: stream has been reset");
-  active_request_ = nullptr;
   onError();
 }
 
+const Envoy::StreamInfo::StreamInfo* ExtProcHttpClient::getStreamInfo() const {
+  if (active_request_ != nullptr) {
+    return &active_request_->streamInfo();
+  } else {
+    return nullptr;
+  }
+}
+
 void ExtProcHttpClient::onError() {
-  // Cancel if the request is active.
```

Comment:
Looks like there is behavior change in next revision. Can you elaborate  we are removing this line? And still keep cancel() in the code base.

Reply:
That's a good question. 

However, there is no logic change here.  The cancel() call is dead code here as from the context, the onError() is always called after active_request_ is set to nullptr. And with active_request_ == nullptr, cancel() is no-op.  

BTW, cancel() call is only needed in the context when the ext_proc filter is destroyed: https://github.com/envoyproxy/envoy/blob/850c7f10cbf5e1132de4eed6b5131d2928ddb95f/source/extensions/filters/http/ext_proc/ext_proc.cc#L474

Another thing is that this PR is trying to move the "active_request_ = nullptr" to after callbacks_->onComplete() and callbacks_->onError() call finished, as these calls will use active_request_ to get the stream_info.

Reply:
OK then original ext_proc http code has redundancy that is cleaned up here